comparative_benchmark/xla_hlo/run_benchmarks.py

```diff
@@ -0,0 +1,358 @@
+  """Returns a list of latencies in milliseconds parsed from XLA logs."""
+  start_matches = GPU_LATENCY_START_REGEXP.findall(raw_output)
+  stop_matches = GPU_LATENCY_STOP_REGEXP.findall(raw_output)
+
+  if len(start_matches) != len(stop_matches):
+    print(
+        f"Error: Unequal number of start and stop logs. {len(start_matches)} start logs != {len(stop_matches)} stop logs."
+    )
+    return []
+
+  if len(start_matches) != expected_iterations:
+    print(
+        f"Error: Number of iterations not equal to the number of expected iteration. Expected {expected_iterations}. Found {len(start_matches)}."
+    )
+    return []
+
+  latencies = [
+      _parse_log_elapsed_time(t1, t2)
+      for t1, t2 in zip(start_matches, stop_matches)
+  ]
+  return latencies
+
+
+def _parse_log_duration(time_str: str) -> float:
+  """Returns the time in milliseconds parsed from XLA logs."""
+  match = TIME_REGEXP.search(time_str)
+  assert match, "Unable to parse the time on log line"
+  exp = TIME_UNITS[match.group(2)]
+  return float(match.group(1)) * exp
+
+
+def _parse_log_size(size_str: str) -> float:
+  """Returns the size in bytes parsed from XLA logs."""
+  match = SIZE_REGEXP.search(size_str)
+  assert match, "Unable to parse the size on log line"
+  return float(match.group(1)) * 1e-6
+
+
+def _parse_gpu_compile_time(raw_output: str) -> float:
+  matches = GPU_COMPILE_TIME_REGEXP.findall(raw_output)
+  total_compile_time_ms = sum([_parse_log_duration(t1) for t1 in matches])
+  return total_compile_time_ms * 1e-3
+
+
+def _parse_gpu_peak_memory(raw_output: str) -> float:
+  matches = GPU_PEAK_MEMORY_REGEXP.findall(raw_output)
+  assert matches, "Unable to find peak memory"
+  return _parse_log_size(matches[-1])
+
+
+def _run_compiler_benchmark_gpu(
+    hlo_benchmark_tool_path: pathlib.Path,
+    hlo_input_path: pathlib.Path,
+    benchmark_iterations: int,
+    verbose: bool,
+) -> Dict[str, Any]:
+  cmd = [
+      hlo_benchmark_tool_path,
+      f"--hlo_file={hlo_input_path}",
+      f"--device_type=gpu",
+      f"--num_repeats={benchmark_iterations}",
+      "--input_format=text",
+      "--num_replicas=1",
+      "--num_partitions=1",
+      "--logtostderr",
+  ]
+  if verbose:
+    print(f"Run command: {cmd}")
+  result = subprocess.run(
+      cmd,
+      stdout=subprocess.PIPE,
+      stderr=subprocess.STDOUT,
+      # Timings are logged under VLOG so we need to enable this for the modules
+      # we are intereated in.
+      env={
+          "TF_CPP_MIN_LOG_LEVEL":
+              "0",
+          "TF_CPP_VMODULE":
+              "nvptx_compiler=1,gpu_compiler=1,parse_flags_from_env=1,bfc_allocator=2,functional_hlo_runner=1",
+      })
+  result_text = result.stdout.decode("utf-8")
+
+  latencies = _parse_gpu_latencies(result_text, benchmark_iterations)
+  compile_time_s = _parse_gpu_compile_time(result_text)
+  peak_memory_usage = _parse_gpu_peak_memory(result_text)
+
+  results_dict = {
+      "compile_time_s": compile_time_s,
```

Comment:
Should we record compile time in `ms` like the rest of the latencies? I initially decided to use seconds because of how long it takes to compile but it might be confused for ms since the rest of the results are in ms.

Reply:
Done.